docs/source/howto/run_codes.rst

```diff
@@ -357,24 +357,215 @@ After :ref:`setting up your computer <how-to:run-codes:computer>` and :ref:`sett
 
  * Submit your calculation to the AiiDA daemon:
 
-   .. code-block:: bash
+    .. code-block:: bash
 
-       verdi run submit.py
+        verdi run submit.py
 
 After this, use ``verdi process list`` to monitor the status of the calculations.
 
 See :ref:`topics:processes:usage:launching` and :ref:`topics:processes:usage:monitoring` for more details.
 
-.. todo::
 
-    .. _how-to:run-codes:caching:
 
-    title: Using caching to save computational resources
+.. _how-to:run-codes:caching:
+
+How to save computational resources using caching
+=================================================
+
+There are numerous reasons why you may need to re-run calculations you’ve already done before.
+Maybe you run a great number of complex workflows in high-throughput that each may repeat the same calculation, or you may have to restart an entire workflow that failed somewhere half-way through.
+Since AiiDA stores the full provenance of each calculation, it can detect whether a calculation has been run before and, instead of running it again, simply reuse its outputs, thereby saving valuable computational resources.
+This is what we mean by **caching** in AiiDA.
+
+.. _how-to:run-codes:caching:enable:
+
+How to enable caching
+---------------------
+
+Caching is **not enabled by default**.
```

Comment:
Should we provide a very brief explanation as to why not or possibly just link to the "Limitations"?

Reply:
There are two reasons, but not sure if we want to mention (any) of them
1. It is quite an unusual feature and if you are not aware of it, it can surprise users a lot.
2. _Especially_, if it does not always do the "correct" thing. This is not even talking about bugs, which would be bad, but for example, if you have a calculation that finished with a non-zero exit status, it _is_ a valid cache source and so will be used. But a plugin may have intended this to really not be cached from. So long story short, it is a bit complicated and we feel currently that users should be aware of what they are enabling.

Reply:
I understand your explanation, but I strongly disagree with your conclusion that we should not mention this, see also my comment on the disclaimer below. The fact that caching is not reliable and false cache hits are expected, makes it even more important to mention this here.

Reply:
Fair, enough. I will add an explanation.